[Java] Implement concept exercise conditionals-ternary (#2265)
exercism/v3: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md

In this exercise, you'll be implementing rules for calculating the total salary of a empyoyee in a month. The International Siderurgy Company (ISC) needs help to calculate the salary for the employees, given that different factors can alter the final wage value for each employee.

You have the following tasks

> ## 1. Decide a penalty for days skipped needs to be applied

Reviewer: Maybe rephrase this slightly as students are returning a multiplier, and not a penalty.

```suggestion
## 1. Determine the salary multiplier
```

Author: Makes sense, while writing i was really thinking that `penalty` was a strong term, but couldn't find a nicer one.

> Implement the `multiplierPerDaysSkipped` method that returns the penalty multiplier based on the days the employee skipped the job. A 15% penalty is applied if more than five days wore skipped.

Reviewer: ```suggestion
Implement the `multiplierPerDaysSkipped` method that returns the salary multiplier based on the number of days the employee skipped the job. A 15% penalty is applied if more than five days were skipped.
```

```java
multiplierPerDaysSkipped(3);
// => 1

multiplierPerDaysSkipped(7);
// =>0.85
```

## 2. Calculate the bonus for products sold

Implement the `multiplierPerProductsSold` and `bonusProductSold` methods. The ISC pays ten monetary units for each product sold, but if the employee sold more than twenty products, the multiplier is improved to thirteen. `multiplierPerProductsSold` should decide which multiplier is applied and `bonusProductSold` should return the total bonus in monetary units.

```java
multiplierPerProductsSold(25);
// => 13

> bonusProductSold(5);

Reviewer: Maybe rename `bonusProductSold` to `bonusForProductsSold`? Just to be a little bit more descriptive.

Author: Agreed

// => 50
```

## 3. Calculate the final salary for the employee

> Implement the `finalSalary` method. It should be able to multiply the base salary by the penalty and sum the bonus and return the result, but keep in mind that salaries should be capped at the maximum value

Reviewer: > multiply the base salary by the penalty

Maybe make this `multiply the base salary by the salary multiplier`? See my previous comment above,

Reviewer: > keep in mind that salaries should be capped at the maximum value

I don't see the actual maximum value listed anywhere. Could you add that?

```java

> finalSalary(0,0);

Reviewer: ```suggestion
finalSalary(0, 0);
```

> finalSalary(0,0);

Reviewer: Which parameter represents which value? I would suggest being explicit about the parameter values in both the instructions above and in this bit of code (see [this example](https://github.com/exercism/v3/blob/master/languages/java/exercises/concept/conditionals/.docs/instructions.md#3-implement-the-decision-logic-for-hand-scores-larger-than-20-points)).

// => 1000

> finalSalary(0,100);

Reviewer: ```suggestion
finalSalary(0, 100);
```

// => 2000
```

### Attention

> You should not use _if-else_ statements in this exercise, only the _ternary operator_ is allowed

Reviewer: It will be entirely possible that students will start coding once they've read one instruction. I'd remove this section and either add something like this to each task, or perhaps add it to the "You have the following tasks" line.

Author: I like the idea of adding it to the beginning, maybe something like:
> You have three tasks, but keep in mind that _if/else_ statements should not be used.
or 
>  You have three tasks, use just _ternary operators_  to solve them

In this exercise, you'll be implementing rules for calculating the total salary of a empyoyee in a month. The International Siderurgy Company (ISC) needs help to calculate the salary for the employees, given that different factors can alter the final wage value for each employee.

> You have three tasks, but keep in mind that if/else statements should not be used.

Reviewer: Maybe make it slightly more explicit what is expected:

```suggestion
You have three tasks and you should use the ternary operator instead of if/else statements to implement them.
```

## 1. Determine the salary multiplier

Implement the `multiplierPerDaysSkipped` method that returns the salary multiplier based on the number of days the employee skipped the job. A 15% penalty is applied if more than five days were skipped.

```java

> 
> int daysSkipped = 3;
> multiplierPerDaysSkipped(daysSkipped);
> // => 1
> 
> daysSkipped = 7;
> multiplierPerDaysSkipped(daysSkipped);
> // => 0.85
> 

Reviewer: ```suggestion
int daysSkipped = 3;
multiplierPerDaysSkipped(daysSkipped);
// => 1

daysSkipped = 7;
multiplierPerDaysSkipped(daysSkipped);
// => 0.85
```

```

## 2. Calculate the bonus for products sold

Implement the `multiplierPerProductsSold` and `bonusForProductSold` methods. The ISC pays ten monetary units for each product sold, but if the employee sold more than twenty products, the multiplier is improved to thirteen. `multiplierPerProductsSold` should decide which multiplier is applied and `bonusForProductSold` should return the total bonus in monetary units.

```java

> 

Reviewer: ```suggestion
```

int productsSold = 21;
multiplierPerProductsSold(productsSold);
// => 13

productsSold = 5;
bonusForProductSold(productsSold);
// => 50
```

## 3. Calculate the final salary for the employee

Implement the `finalSalary` method. It should be able to multiply the base salary of 1000.00 by the salary multiplier and sum the bonus and return the result, but keep in mind that salaries should be capped at 2000.00;

```java

> 
> int daysSkipped = 2;
> int productsSold = 3;
> finalSalary(daysSkipped, productsSold);
> // => 1030
> 

Reviewer: ```suggestion
int daysSkipped = 2;
int productsSold = 3;
finalSalary(daysSkipped, productsSold);
// => 1030
```
